src/widgets/dlicensedialog.cpp

```diff
@@ -0,0 +1,236 @@
+    fontManager->bind(componentVersionLabel, DFontSizeManager::T6, QFont::DemiBold);
+    fontManager->bind(copyRightLabel, DFontSizeManager::T6, QFont::DemiBold);
+
+    licenseContentLabel->setWordWrap(true);
+
+    QWidget *licenseWidget = new QWidget;
+    QVBoxLayout *licenseLayout = new QVBoxLayout(licenseWidget);
+    licenseLayout->setSpacing(0);
+    licenseLayout->setMargin(20);
+    licenseLayout->addWidget(componentNameLabel);
+    licenseLayout->addSpacing(16);
+    licenseLayout->addWidget(new DHorizontalLine);
+    licenseLayout->addSpacing(16);
+    licenseLayout->addWidget(componentVersionLabel);
+    licenseLayout->addWidget(copyRightLabel);
+    licenseLayout->addSpacing(40);
+    licenseLayout->addWidget(licenseContentLabel);
+    licenseLayout->addStretch(0);
+
+    scrollArea->setFrameStyle(QFrame::NoFrame);
+    scrollArea->viewport()->setAutoFillBackground(false);
+    scrollArea->setContentsMargins(QMargins(0,0,0,0));
+    scrollArea->viewport()->setContentsMargins(QMargins(0,0,0,0));
+    scrollArea->setWidget(licenseWidget);
+    scrollArea->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+    scrollArea->setWidgetResizable(true);
+    licenseWidget->setAutoFillBackground(false);
+
+    stackedLayout->addWidget(listView);
+    stackedLayout->addWidget(scrollArea);
+
+    QVBoxLayout *mainLayout = new QVBoxLayout;
+    mainLayout->setContentsMargins(10, 0, 10, 0);
+    mainLayout->addWidget(titleBar, 0, Qt::AlignTop);
+    mainLayout->addLayout(stackedLayout);
+
+    q->setLayout(mainLayout);
+    q->setFocusPolicy(Qt::ClickFocus);
+    q->setFocus();
+
+    QObject::connect(backwardBtn, &QAbstractButton::clicked, q, [this]{
+        backwardBtn->setVisible(false);
+        scrollArea->horizontalScrollBar()->setValue(0);
+        scrollArea->verticalScrollBar()->setValue(0);
+        stackedLayout->setCurrentIndex(0);
+    });
+    QObject::connect(listView, &QAbstractItemView::clicked, q, [this](const QModelIndex &index) {
+        auto item = listModel->item(index.row());
+        DLicenseInfo::DComponentInfo *componentInfo;
+        for (const auto &info : licenseInfo.componentInfos()) {
+            if (info->name() == item->text()) {
+                componentInfo = info;
+                break;
+            }
+        }
+        if (!backwardBtn->isVisible())
+            backwardBtn->setVisible(true);
+        componentNameLabel->setText(componentInfo->name());
+        componentVersionLabel->setText(componentInfo->version());
+        copyRightLabel->setText(componentInfo->copyRight());
+        licenseContentLabel->setText(licenseInfo.licenseContent(componentInfo->licenseName()));
+        stackedLayout->setCurrentIndex(1);
+    });
+}
+
+void DLicenseDialogPrivate::addComponentItem(const DLicenseInfo::DComponentInfo *componentInfo)
+{
+    D_Q(DLicenseDialog);
+    auto pItem = new DStandardItem(componentInfo->name());
```

Comment:
```suggestion
    auto pItem = new DStandardItem(componentInfo->name());
    pItem->setData(componentInfo);
```
这样就不需要 clicked 时再去遍历了，不过要主要清理后要把item这个 data 清理， 可以用 QPointer

Reply:
done